packages/bot-skeleton/src/scratch/blocks/Binary/Trade Definition/trade_definition_tradeoptions.js

```diff
@@ -68,6 +69,20 @@ Blockly.Blocks.trade_definition_tradeoptions = {
         };
     },
     onchange(event) {
+        if (event.type === 'change') {
+            const selected_block = this.workspace.getBlockById(event.blockId);
```

Comment:
I think you can get the value by this line of code: 
Blockly.JavaScript.valueToCode(field_name)

Reply:
How about setting the value? Is there a similar code to set the value as simple as this?
Because after getting the value we need to update it if needed.

Reply:
selected_block.setFieldValue()  should work I think

Reply:
It didn't :cry: